File: backend/workspace/test_debug/buggy_code.py
"""Test file with intentional bugs for debugging demo"""

def calculate_average(numbers):
    """Calculate average - has division by zero bug"""
    # BUG: No check for empty list
    total = sum(numbers)
    return total / len(numbers)  # Will crash on empty list!

def process_data(data):
    """Process data - has type error"""
    # BUG: Assumes data is always a dict
    return data['value'] * 2  # Will crash if data is not a dict!

def find_max(items):
    """Find maximum - has edge case bug"""
    # BUG: Doesn't handle empty list
    max_val = items[0]  # IndexError on empty list!
    for item in items:
        if item > max_val:
            max_val = item
    return max_val

class DataProcessor:
    def __init__(self):
        # BUG: Typo in attribute name
        self.procesor_count = 0  # Should be processor_count
    
    def process(self, data):
        # BUG: Using undefined attribute
        self.processor_count += 1  # AttributeError!
        return data * 2

# Test code that will fail
if __name__ == "__main__":
    # This will cause errors when run
    print("Testing buggy code...")
    
    # Test 1: Division by zero
    result = calculate_average([])
    print(f"Average: {result}")
    
    # Test 2: Type error
    result = process_data("not a dict")
    print(f"Processed: {result}")
    
    # Test 3: Index error
    result = find_max([])
    print(f"Max: {result}")

File: backend/workspace/test_debug/buggy_code.js
// Test file with JavaScript bugs

function divideNumbers(a, b) {
    // BUG: No check for division by zero
    return a / b;
}

function getProperty(obj, prop) {
    // BUG: No null check
    return obj[prop].toUpperCase();  // Will crash if obj is null or prop doesn't exist
}

async function fetchData(url) {
    // BUG: No error handling
    const response = await fetch(url);
    const data = await response.json();  // Will crash on non-JSON response
    return data;
}

class Calculator {
    constructor() {
        // BUG: Typo in property
        this.resuIt = 0;  // Should be result
    }
    
    add(a, b) {
        // BUG: Using wrong property name
        this.result = a + b;  // ReferenceError!
        return this.result;
    }
}

// Test code
console.log("Testing buggy JavaScript...");
console.log(divideNumbers(10, 0));  // Infinity, but should handle better
console.log(getProperty(null, "test"));  // Will crash